### PR TITLE
FEATURE: Add a site setting to allow emojis to come from an external URL

### PR DESCRIPTION
**Pull request by xfalcox**

### CODE REVIEW

File: app/assets/javascripts/pretty-text/engines/discourse-markdown/emoji.js [hunk 1]
```diff
@@ -322,6 +322,8 @@ export function setup(helper) {
     opts.features.inlineEmoji = !!siteSettings.enable_inline_emoji_translation;
     opts.emojiSet = siteSettings.emoji_set || "";
     opts.customEmoji = state.customEmoji;
+    opts.enableEmojiCDN = siteSettings.external_emoji_url_enabled;
```
Comment (SamSaffron) [inline app/assets/javascripts/pretty-text/engines/discourse-markdown/emoji.js:325]: I don't think we need 2 settings here, it just grows our json paylod for no huge benefit. A single `external_emoji_url` can work. If it is blank then just assume not using external.

**Comment by riking** [comment]
Should this go into a `setupData` instead of a client site setting, so that it could potentially be the same as S3CDN / SITE_CDN on self-hosted sites that have that?

**Comment by SamSaffron** [comment]
@riking I think we get that for free anyway, cause we are using getURL etc internally.

**Comment by SamSaffron** [comment]
Test failures look legit @xfalcox 

```
127.0.0.1 - - [23/Feb/2021:22:42:00 +0000] "GET /undefined/emoji_one/joy.png?v=9 HTTP/1.1" 404 - 0.0121
127.0.0.1 - - [23/Feb/2021:22:42:00 +0000] "GET /undefined/emoji_one/relaxed.png?v=9 HTTP/1.1" 404 - 0.0155
127.0.0.1 - - [23/Feb/2021:22:42:00 +0000] "GET /undefined/emoji_one/wink.png?v=9 HTTP/1.1" 404 - 0.0264
127.0.0.1 - - [23/Feb/2021:22:42:00 +0000] "GET /undefined/emoji_one/neutral_face.png?v=9 HTTP/1.1" 404 - 0.01
```

I am guessing something is not set right somewhere.

**Comment by xfalcox** [comment]
> Test failures look legit @xfalcox
> 
> ```
> 127.0.0.1 - - [23/Feb/2021:22:42:00 +0000] "GET /undefined/emoji_one/joy.png?v=9 HTTP/1.1" 404 - 0.0121
> 127.0.0.1 - - [23/Feb/2021:22:42:00 +0000] "GET /undefined/emoji_one/relaxed.png?v=9 HTTP/1.1" 404 - 0.0155
> 127.0.0.1 - - [23/Feb/2021:22:42:00 +0000] "GET /undefined/emoji_one/wink.png?v=9 HTTP/1.1" 404 - 0.0264
> 127.0.0.1 - - [23/Feb/2021:22:42:00 +0000] "GET /undefined/emoji_one/neutral_face.png?v=9 HTTP/1.1" 404 - 0.01
> ```
> 
> I am guessing something is not set right somewhere.

I was aware and chasing this for some long hours :sleepy:

**Comment by riking** [comment]
Add a test that `download_remote_images` will ignore these?

**Comment by xfalcox** [comment]
> Add a test that `download_remote_images` will ignore these?

Added on 807471d

**Review by riking — APPROVED** [review]

**Pull request merged**